src/nvim/ops.c

```diff
@@ -2835,8 +2835,11 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
     curr->y_size = j;
     xfree(reg->y_array);
   }
-  if (curwin->w_p_rnu) {
-    redraw_later(curwin, SOME_VALID);  // cursor moved to start
+  if (curwin->w_p_rnu && yanklines > 1) {
+    // TODO(lewis6991): for some unknown reason if 'cursorline' is set,
+    // redrawing with VALID here breaks and we must use SOME_VALID.
+    // See test 'always updated. vim-patch:8.1.0849' in test/functional/ui/highlight_spec.lua
+    redraw_later(curwin, curwin->w_p_cul ? SOME_VALID : VALID);  // cursor moved to start
```

Comment:
I've figured out something about this redrawing problem. Using `VALID` here causes `update_screen()` to be called before `validate_cursor()`, and `update_screen()` adds `VALID_WCOL` to `wp->w_valid`, which prevents `curs_rows()` from being called later.

Reply:
Perhaps this call to `redraw_later()` isn't necessary at all?

Reply:
Oh yes, this call is unnecessary after patch 8.1.0374

Reply:
<https://github.com/vim/vim/pull/9971>